packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -157,6 +164,98 @@ const methods = {
     return result
   },
 
+  /**
+   * fetch XS >= 8.4 updates
+   * @param {Host} host
+   * @returns {Promise<{
+   * hosts: Array<{
+   *  ref: String,
+   *  guidance: {
+   *    mandatory: Array<String>,
+   *    recommended: Array<String>,
+   *    full: Array<String>
+   *  },
+   *  RPMS: Array<String>,
+   *  updates: Array<String>,
+   *  livepatches: Array<{
+   *    component: String,
+   *    base_build_id: String,
+   *    base_release: String,
+   *    to_version: String,
+   *    to_release: String
+   *  }>
+   * }>,
+   * updates: Array<{
+   *  id: String,
+   *  summary: String,
+   *  description: String,
+   *  "special-info": String,
+   *  URL: String,
+   *  type: String,
+   *  issued: String,
+   *  severity: String,
+   *  livepatches: Array,
+   *  guidance: {
+   *    mandatory: Array<String>,
+   *    recommended: Array<String>,
+   *    full: Array<String>
+   *  },
+   *  title: String,
+   * }>,
+   * hash: String
+   * }>}
+   */
+  async _fetchXsUpdatesEndpoint(host) {
+    const { xo } = this
+
+    const serverId = xo.getXenServerIdByObject({ $pool: host.$pool.uuid, id: host.uuid })
+    const server = await xo.getXenServerWithCredentials(serverId)
+
+    const url = new URL(`http://${server.host}/updates`)
+    url.protocol = this._url.protocol
+    const opts = {
+      headers: { Authorization: `Basic ${Buffer.from(`${server.username}:${server.password}`).toString('base64')}` },
+      rejectUnauthorized: !this._allowUnauthorized,
+    }
+
+    const resp = await xo.httpRequest(url, opts)
+    return resp.json()
+  },
+
+  // List updates for XS >= 8.4
+  async _listXsUpdates(host) {
+    const xsUpdates = await this._fetchXsUpdatesEndpoint(host)
+    const updateById = keyBy(xsUpdates.updates, 'id')
+
+    const updatesInfo = xsUpdates.hosts.find(_host => _host.ref === host.$ref)
+    if (updatesInfo === undefined) {
+      throw new Error(`Host ref: ${host.$ref} match no hosts`)
+    }
+
+    const hostUpdates = updatesInfo.updates.map(updateId => {
+      const update = updateById[updateId]
+      const guidances = Array.from(new Set(Object.values(update.guidance).flat())).join(', ')
```

Comment:
That's a lot of parenthesis on the same line, it's still understandable but a bit hard to read. Maybe we could split it into two lines, or add a short comment?